### PR TITLE
feat(DATAGO-120931): Gateway Auto-Registration via A2A Discovery

### PR DESCRIPTION
**Pull request by mo-radwan1**
# Gateway Auto-Registration and Health Tracking via A2A Discovery

## Overview

Enables automatic registration and health tracking of all gateways through the A2A discovery protocol, allowing platform services to monitor gateway deployment status and fleet health without requiring code modifications in individual gateway implementations.

### 1. Gateway Card Publishing (Community)

**What:** All gateways automatically publish discovery cards as heartbeats
- REST, Slack, Teams, and HTTP SSE gateways auto-register on startup
- Periodic publishing every 30 seconds (configurable)
- Zero configuration required for basic functionality
- Operators can customize or disable via configuration

**How:** Discovery cards include gateway-specific extensions:
- Gateway-role extension: gateway_id, gateway_type, namespace
- Deployment extension (optional): deployment_id for k8s/container tracking

### 2. Gateway Registry (Community) ⭐ NEW

**What:** GatewayRegistry tracks all discovered gateways with health monitoring
- Thread-safe in-memory tracking of gateway discovery cards
- Health status based on heartbeat TTL (default 90 seconds)
- Automatic filtering of healthy vs unhealthy gateways
- Metadata extraction: gateway_type, namespace, deployment_id
- Lifecycle callbacks for gateway add/remove events

**Architecture:** Parallel implementation to AgentRegistry
- Gateways publish cards → Discovery topic → Components route → GatewayRegistry tracks
- Same thread-safe patterns and health monitoring as AgentRegistry
- Coexists with agent discovery (no conflicts)

**Available in:**
- BaseGatewayComponent (all gateways track peer gateways)
- PlatformServiceComponent (platform service monitors all gateways)

### 3. Discovery Card Routing (Community) ⭐ NEW

**What:** Automatic routing of discovery messages to appropriate registries
- Gateway cards (with gateway-role extension) → GatewayRegistry
- Agent cards → AgentRegistry (existing behavior preserved)
- Logging for gateway events: INFO for new discovery, DEBUG for heartbeats

**Implementation:**
- BaseGatewayComponent._handle_discovery_message()
- PlatformServiceComponent._handle_discovery_message()
- Uses is_gateway_card() utility to identify card type
- Routes to correct registry
- No breaking changes to existing agent discovery

### 4. Platform Service Gateway Monitoring (Community) ⭐ NEW

**What:** HTTP endpoints for gateway fleet health monitoring (admin/ops APIs)

```
GET /api/v1/platform/gateways
  - Returns list of discovered gateway AgentCards
  - Sorted by gateway_id
  - Platform control plane API

GET /api/v1/platform/gateways/health
  - Health status for all gateways
  - Returns healthy/unhealthy counts
  - Includes gateway metadata (type, namespace, deployment_id)
  - Configurable TTL threshold

GET /api/v1/platform/gateways/{gateway_id}/health
  - Health status for specific gateway
  - Returns 404 if gateway not found
  - Includes last_seen timestamp and seconds since last heartbeat
```

**Architecture:** Platform Service is the control plane
- Platform Service monitors infrastructure (deployers, agents, gateways)
- Separates admin/ops APIs from user-facing APIs
- WebUI Gateway remains focused on user features (chat, agent discovery)

### Gateway Card Publishing (Optional)

Three new optional fields in gateway app_config:

```yaml
app_config:
  # Gateway type (auto-detected if not specified)
  gateway_type: http_sse  # or slack, rest, teams, generic

  # Gateway card publishing configuration
  gateway_card_publishing:
    enabled: true           # Enable/disable publishing (default: true)
    interval_seconds: 30    # Heartbeat interval (default: 30)

  # Gateway card metadata
  gateway_card:
    description: "Production HTTP Gateway"  # Custom description

  # Deployment tracking (optional)
  deployment:
    id: k8s-pod-abc123     # Links to deployment system
```

### Files Created

**Core Implementation:**
- `src/solace_agent_mesh/common/gateway_registry.py` (280 lines)
  - GatewayRegistry class with health tracking
- `src/solace_agent_mesh/common/a2a/utils.py` (90 lines)
  - is_gateway_card() and extract_gateway_info() helpers
- `src/solace_agent_mesh/services/platform/api/routers/gateway_router.py` (220 lines)
  - Platform Service gateway monitoring endpoints

**Tests:**
- `tests/unit/common/test_gateway_registry.py` (440 lines, 48 tests)
- `tests/unit/common/a2a/test_utils.py` (260 lines, 14 tests)
- `tests/unit/services/platform/routers/test_gateway_router.py` (270 lines, 11 tests)

**Total:** ~1,560 lines of new code, 73 tests, 100% passing

### Files Modified

**Gateway Components:**
- `src/solace_agent_mesh/gateway/base/app.py` - Gateway card config schema
- `src/solace_agent_mesh/gateway/base/component.py` - Integrate GatewayRegistry

**Platform Service:**
- `src/solace_agent_mesh/services/platform/component.py` - Add GatewayRegistry tracking
- `src/solace_agent_mesh/services/platform/api/dependencies.py` - Add dependency injection
- `src/solace_agent_mesh/services/platform/api/routers/__init__.py` - Register router

**Common:**
- `src/solace_agent_mesh/common/a2a/__init__.py` - Export new utilities

## Architecture

```
┌─────────────────────────────────────────────────────────────────┐
│                         Community Repo                           │
│                                                                  │
│  ┌──────────────────┐          ┌──────────────────┐            │
│  │  AgentRegistry   │          │ GatewayRegistry  │            │
│  │                  │          │  (NEW)           │            │
│  │ - Track agents   │          │ - Track gateways │            │
│  │ - Health checks  │          │ - Health checks  │            │
│  │ - TTL monitoring │          │ - TTL 90s        │            │
│  └────────▲─────────┘          └────────▲─────────┘            │
│           │                             │                       │
│           │     Discovery Topic         │                       │
│           │     Subscription            │                       │
│           └────────┬────────────────────┘                       │
│                    │                                            │
│     ┌──────────────┴──────────────┐                            │
│     │                              │                            │
│  BaseGatewayComponent    PlatformServiceComponent              │
│  - Routes discovery      - Routes discovery                    │
│  - Publishes own card    - Monitors infrastructure             │
│  - Tracks peer gateways  - Exposes health APIs                 │
│                                                                 │
│  APIs:                                                          │
│  ┌─────────────────────────────────────────────────┐          │
│  │ WebUI Gateway (User-facing)                     │          │
│  │   GET /api/v1/agentCards   (discover agents)    │          │
│  └─────────────────────────────────────────────────┘          │
│                                                                 │
│  ┌─────────────────────────────────────────────────┐          │
│  │ Platform Service (Admin/Ops)                    │          │
│  │   GET /api/v1/platform/health                   │          │
│  │   GET /api/v1/platform/gateways        (NEW)    │          │
│  │   GET /api/v1/platform/gateways/health (NEW)    │          │
│  └─────────────────────────────────────────────────┘          │
│                                                                 │
└─────────────────────────────────────────────────────────────────┘
```

### Design Principle: Separation of Concerns

**WebUI Gateway** (port 8080)
- Purpose: User application gateway
- Users: End users interacting with SAM
- APIs: Chat, tasks, agent discovery, sessions, artifacts

**Platform Service** (port 8081 or embedded)
- Purpose: Control plane for infrastructure management
- Users: Platform operators, deployment systems
- APIs: Agent deployments, deployer heartbeats, gateway health

### Unit Tests (73 tests)

**GatewayRegistry Tests (48 tests):**
- Basic operations: add, update, get, remove, clear
- Health tracking: TTL expiration, last_seen timestamps
- Metadata extraction: gateway_type, namespace, deployment_id
- Thread safety: concurrent operations without deadlocks
- Edge cases: empty names, rapid updates, data cleanup
- Callbacks: lifecycle events with error handling

**Utility Function Tests (14 tests):**
- Gateway card identification (is_gateway_card)
- Metadata extraction (extract_gateway_info)
- Edge cases (None, missing extensions)

**Platform Service API Tests (11 tests):**
- GET /platform/gateways endpoint
- GET /platform/gateways/health endpoint
- GET /platform/gateways/{id}/health endpoint
- Error handling (404 for missing, 503 for unavailable registry)

### Test Results

```bash
pytest tests/unit/common/test_gateway_registry.py \
       tests/unit/common/a2a/test_utils.py \
       tests/unit/services/platform/routers/test_gateway_router.py -v

Result: 73 passed in 6.75s (100% pass rate)
```

### Integration Testing

Live testing script included to verify complete flow:
- Gateway registry operations
- Health monitoring and TTL expiration
- Metadata extraction
- Lifecycle callbacks

### 1. Gateway Fleet Monitoring (Platform Operators)
Query all discovered gateways and their health status:
```bash
curl http://localhost:8081/api/v1/platform/gateways/health | jq
```

**Response:**
```json
{
  "total_gateways": 3,
  "healthy_count": 2,
  "unhealthy_count": 1,
  "ttl_seconds": 90,
  "gateways": [
    {
      "gateway_id": "http-gateway-prod-1",
      "gateway_type": "http_sse",
      "namespace": "prod/mycompany",
      "deployment_id": "k8s-pod-abc123",
      "health_status": "healthy",
      "last_seen": 1704567890.123,
      "seconds_since_last_seen": 5
    }
  ]
}
```

### 2. Deployment Status Tracking
Monitor gateways by deployment ID for k8s pod tracking:
```javascript
const response = await fetch('/api/v1/platform/gateways/health');
const deployment = response.gateways
  .filter(g => g.deployment_id === 'k8s-deployment-123');
```

### 3. Health Dashboards
Build operator dashboards showing gateway fleet health:
- Real-time heartbeat monitoring
- Stale gateway detection (>90s since last heartbeat)
- Gateway distribution across namespaces
- Gateway type breakdown (http_sse, slack, rest, teams)

### 4. Automated Failover
Detect and react to gateway failures:
- Monitor heartbeat TTL via Platform Service
- Trigger alerts when gateways go unhealthy
- Automated traffic redirection

## Backward Compatibility

✅ **Fully backward compatible**
- Existing gateways function without configuration changes
- Gateway card publishing enabled by default but configurable
- Auto-detection of gateway type from class names
- No breaking changes to agent discovery functionality
- WebUI Gateway retains all user-facing features

## Jira Requirements

| Requirement | Status | Implementation |
|-------------|--------|----------------|
| Gateway registry service | ✅ Complete | GatewayRegistry class (community) |
| Subscribe to heartbeat topic | ✅ Complete | Via discovery topic subscription |
| Process heartbeat messages | ✅ Complete | Discovery message routing in components |
| Track last_seen timestamp | ✅ Complete | GatewayRegistry._last_seen |
| Health status tracking | ✅ Complete | check_ttl_expired(), get_healthy/unhealthy_gateways() |
| State transition (first heartbeat) | ✅ Complete | is_new flag + on_gateway_added callback |
| Timeout configuration | ✅ Complete | Configurable TTL (default 90s) |
| Status API integration | ✅ Complete | 3 REST endpoints in Platform Service |
| Background job | ⏸️ Optional | Can add timer for auto-cleanup if needed |

## Example: Multi-Gateway Deployment

```yaml
# Gateway 1: HTTP SSE (Web UI)
gateway_id: web-ui-gateway-prod
gateway_type: http_sse
deployment:
  id: k8s-deployment-web-ui-abc123

# Gateway 2: Slack (Team Alpha)
gateway_id: slack-gateway-team-alpha
gateway_type: slack

# Gateway 3: REST API
gateway_id: rest-api-gateway-v2
gateway_type: rest
deployment:
  id: docker-compose-api-xyz789
```

**Result:** Platform Service tracks all 3 gateways:
- Platform Service `gateway_registry` contains all 3 gateways
- Health status available via `/api/v1/platform/gateways/health`
- Each gateway also tracks peer gateways locally

### Typical Production Deployment

```
┌─────────────────────────────────────────────────┐
│              Platform Service                    │
│        (Control Plane - Port 8081)              │
│                                                  │
│  GET /api/v1/platform/health                    │
│  GET /api/v1/platform/gateways        ← Monitor │
│  GET /api/v1/platform/gateways/health ← Monitor │
│  POST /api/v1/platform/agents                   │
│  POST /api/v1/platform/deployments              │
└─────────────────────────────────────────────────┘
                      ▲
                      │ Discovery
                      │ Heartbeats
                      │
      ┌───────────────┼───────────────┐
      │               │               │
┌─────┴─────┐  ┌─────┴─────┐  ┌─────┴─────┐
│  WebUI    │  │  Slack    │  │   REST    │
│  Gateway  │  │  Gateway  │  │  Gateway  │
│  (8080)   │  │           │  │  (9000)   │
└───────────┘  └───────────┘  └───────────┘
     │              │              │
     └──────────────┴──────────────┘
         Publishing heartbeats
```

## Next Steps (Enterprise)

Future enhancements in enterprise repo:
- Database persistence for historical health metrics
- Advanced deployment management UI
- Alerting and notifications for gateway failures
- Multi-tenancy (gateway isolation per customer)
- Gateway provisioning and configuration APIs
- Integration with k8s operators for automated recovery

### Run Unit Tests
```bash
source .venv/bin/activate

# Test GatewayRegistry
pytest tests/unit/common/test_gateway_registry.py -v

# Test utility functions
pytest tests/unit/common/a2a/test_utils.py -v

# Test Platform Service endpoints
pytest tests/unit/services/platform/routers/test_gateway_router.py -v

# Run all gateway tests
pytest tests/unit/common/test_gateway_registry.py \
       tests/unit/common/a2a/test_utils.py \
       tests/unit/services/platform/routers/test_gateway_router.py -v
```

#### Option 1: With Platform Service

```bash
# Start Platform Service (monitors all gateways)
python -m solace_agent_mesh.services.platform.app -c platform_config.yaml

# Query gateway fleet health
curl http://localhost:8081/api/v1/platform/gateways/health | jq
```

#### Option 2: With WebUI Gateway (peer discovery)

```bash
# Start WebUI Gateway (tracks peer gateways)
python -m solace_agent_mesh.gateway.http_sse.main -c gateway_config.yaml

# Gateway discovers peer gateways via discovery topic
# Access via component.gateway_registry (programmatic)
```

## Commits

1. `feat`: Add GatewayRegistry for tracking discovered gateways
2. `feat`: Add utility helpers to identify gateway discovery cards
3. `feat`: Integrate GatewayRegistry into BaseGatewayComponent
4. `feat`: Add REST API endpoints for gateway discovery and health monitoring
5. `test`: Add comprehensive unit tests for gateway registry
6. `refactor`: Move gateway health endpoints to Platform Service

## Related

- **Jira:** DATAGO-120931
- **KA Document:** https://sol-jira.atlassian.net/wiki/spaces/MAASB/pages/5663948874
- **Parent Epic:** DATAGO-114468 - Gateway Management in Agent Mesh Enterprise

### CODE REVIEW

File: src/solace_agent_mesh/common/a2a/utils.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    for ext in extensions:
+        ext_uri = ext.uri if hasattr(ext, 'uri') else ext.get('uri')
+        if ext_uri == "https://solace.com/a2a/extensions/sam/gateway-role":
+            return True
+
+    return False
+
+
+def extract_gateway_info(agent_card: AgentCard) -> Optional[Dict[str, Any]]:
+    """
```
Comment (efunneko) [inline src/solace_agent_mesh/common/a2a/utils.py:50]: I am not necessarily asking for a separate function for 'extract_agent_info', but we might consider adding it here if it simplifies code elsewhere for the agents

File: src/solace_agent_mesh/common/gateway_registry.py [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+import time
+from typing import Dict, List, Optional, Tuple
+import logging
+
+from a2a.types import AgentCard
+
+log = logging.getLogger(__name__)
+
+
+class GatewayRegistry:
```
Comment (efunneko) [inline src/solace_agent_mesh/common/gateway_registry.py]: I am wondering if we should have a base Registry class that is shared between the agent and gateway registries... Did you consider this?

File: src/solace_agent_mesh/gateway/base/component.py [hunk 3]
```diff
@@ -2082,6 +2117,131 @@ async def _send_error_to_external(
...
+        configured_type = self.get_config("gateway_type")
+        if configured_type:
+            return configured_type
+
+        class_name = self.__class__.__name__
+        if "WebUI" in class_name or "HttpSse" in class_name:
+            return "http_sse"
+
+        if hasattr(self, 'adapter') and self.adapter:
+            adapter_name = self.adapter.__class__.__name__.lower()
```
Comment (efunneko) [inline src/solace_agent_mesh/gateway/base/component.py:2311]: Q: why do we need to do this and should this be config in the adapter that we just forward?

File: src/solace_agent_mesh/services/platform/api/routers/gateway_router.py [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+        log.exception("%sError retrieving gateways: %s", log_prefix, e)
+        raise HTTPException(
+            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            detail="Internal server error retrieving gateway list"
+        )
+
+
+@router.get("/gateways/health", tags=["Gateways"])
+async def get_gateways_health(
+    gateway_registry: GatewayRegistry = Depends(get_gateway_registry),
```
Comment (efunneko) [inline src/solace_agent_mesh/services/platform/api/routers/gateway_router.py]: How does this align with the work that Greg is doing for health checks for K8S that he is looking at right now? If you haven't chatted to him about this, you should just make sure we aren't overlapping

**Review by efunneko — CHANGES_REQUESTED** [review]
Perhaps nothing really needs to be changed, but I would like answers to my questions first

File: src/solace_agent_mesh/services/platform/api/routers/gateway_router.py [hunk 5]
```diff
@@ -0,0 +1,225 @@
...
+            status_code=status.HTTP_503_SERVICE_UNAVAILABLE,
+            detail="Gateway registry not initialized"
+        )
+
+    try:
+        gateway_ids = gateway_registry.get_gateway_ids()
+        gateways = [
+            gateway_registry.get_gateway(gid)
+            for gid in gateway_ids
+            if gateway_registry.get_gateway(gid)
```
Comment (efunneko) [inline src/solace_agent_mesh/services/platform/api/routers/gateway_router.py]: Probably not a big deal since we don't have too many gateways, but this is calling get_gateway twice for each iteration

File: src/solace_agent_mesh/services/platform/api/routers/gateway_router.py (hunk 4 above)
Comment (gregmeldrum) [reply src/solace_agent_mesh/services/platform/api/routers/gateway_router.py]: We've chatted and there is no/minimal overlap. Mo's checks are for deployment health, including distributed components, and the K8S check are health of the individual containers.

File: src/solace_agent_mesh/common/a2a/utils.py (hunk 1 above)
Comment (mo-radwan1) [reply src/solace_agent_mesh/common/a2a/utils.py:50]: Good point. I looked into this and the difference is that agent info (name, description, skills, etc.) is directly accessible from AgentCard properties, whereas gateway info needs to be extracted from extensions.

I searched for duplicated agent extension parsing code and didn't find any - most places just use agent_card.name directly. So there's nothing to simplify right now.

If we start parsing agent extensions in multiple places later, we can add extract_agent_info() then.

File: src/solace_agent_mesh/common/gateway_registry.py (hunk 2 above)
Comment (mo-radwan1) [reply src/solace_agent_mesh/common/gateway_registry.py]: Yes, good call. I implemented this - created a BaseRegistry class that both AgentRegistry and GatewayRegistry now extend.

The shared functionality includes:
  - Thread-safe storage with locking
  - TTL checking for health monitoring
  - Lifecycle callbacks (on_added, on_removed)
  - Common methods (add_or_update, get, remove, clear, etc.)

GatewayRegistry keeps the gateway-specific extension extraction methods (get_gateway_type, get_gateway_namespace, get_deployment_id) since those are unique to gateways.

File: src/solace_agent_mesh/gateway/base/component.py (hunk 3 above)
Comment (mo-radwan1) [reply src/solace_agent_mesh/gateway/base/component.py:2311]: Config is actually supported and takes priority - see line 2122-2124. The auto-detection is just a fallback for convenience, so users don't have to configure it explicitly if the gateway type can be inferred.

  The logic is:
  1. If gateway_type is configured → use it
  2. Otherwise → try to detect from class name or adapter name
  3. If nothing matches → default to "generic"

So adapters can configure it if they want, but they don't have to. Does that answer your question or do you think we should require explicit config?

File: src/solace_agent_mesh/common/base_registry.py [hunk 6]
```diff
@@ -0,0 +1,252 @@
...
+
+    def add_or_update(self, card: AgentCard) -> bool:
+        """
+        Adds a new entity or updates an existing one.
+
+        Args:
+            card: AgentCard representing the entity
+
+        Returns:
+            True if this is a new entity, False if updating existing
```
Comment (bwiebe-solace) [inline src/solace_agent_mesh/common/base_registry.py]: Minor: doc comment doesn't indicate that False could be returned if the card is invalid / missing a name

File: src/solace_agent_mesh/common/base_registry.py [hunk 7]
```diff
@@ -0,0 +1,252 @@
...
+            del self._items[item_id]
+            if item_id in self._last_seen:
+                del self._last_seen[item_id]
+
+            log.info(
+                "%s '%s' successfully removed from registry",
+                self._entity_name.capitalize(),
+                item_id,
+            )
+            removed = True
```
Comment (bwiebe-solace) [inline src/solace_agent_mesh/common/base_registry.py]: Removed is unnecessary since it's only ever set to True and only assigned in this one place. We don't (currently) have a failure case for deletion, once we're sure that the requested object exists, so we'll always be returning True if we get to the end of this method. Either we should remove this unnecessary assignment, or maybe just restructure this if we anticipate having a failure case where we need a flag like this in the future.

File: src/solace_agent_mesh/common/base_registry.py [hunk 8]
```diff
@@ -0,0 +1,252 @@
...
+                )
+                return False
+
+            last_seen_time = self._last_seen.get(item_id)
+            current_time = time.time()
+            time_since_last_seen = (
+                int(current_time - last_seen_time) if last_seen_time else "unknown"
+            )
+
+            log.warning(
```
Comment (bwiebe-solace) [inline src/solace_agent_mesh/common/base_registry.py]: Just confirming, is warning the right level for this log?

File: src/solace_agent_mesh/common/base_registry.py [hunk 9]
```diff
@@ -0,0 +1,252 @@
...
+                    "Error in %s removed callback for %s: %s",
+                    self._entity_name,
+                    item_id,
+                    e,
+                    exc_info=True,
+                )
+
+        return removed
+
+    def clear(self):
```
Comment (bwiebe-solace) [inline src/solace_agent_mesh/common/base_registry.py:220]: Since we're logging the removals above, should we be logging something here when we're completely clearing the entities?

File: src/solace_agent_mesh/gateway/base/component.py [hunk 10]
```diff
@@ -2262,6 +2297,135 @@ async def _send_error_to_external(
...
+            if "rest" in adapter_name:
```
Comment (bwiebe-solace) [inline src/solace_agent_mesh/gateway/base/component.py:2312]: Just a thought, maybe we should maintain a list of "known" types so that we can just check for existence in the list instead of needing to do this if/return chain. That might also make things more maintainable if more gateway types get added (we might be able to re-use that list elsewhere too, not sure)

File: src/solace_agent_mesh/services/platform/component.py [hunk 11]
```diff
@@ -452,13 +460,37 @@ def _handle_discovery_message(self, payload: Dict) -> bool:
             True if processed successfully, False otherwise
         """
         try:
+            from solace_agent_mesh.common.a2a.utils import is_gateway_card
```
Comment (bwiebe-solace) [inline src/solace_agent_mesh/services/platform/component.py]: Suggestion: looks like this import doesn't need to happen here, unless I'm missing something, so it would be cleaner to have it at the top of the file.

File: tests/unit/common/test_gateway_registry.py [hunk 12]
```diff
@@ -0,0 +1,706 @@
...
+import pytest
+import time
+import threading
+from unittest.mock import MagicMock
+from a2a.types import AgentCard, AgentCapabilities, AgentExtension
+
+from solace_agent_mesh.common.gateway_registry import GatewayRegistry
+
+
+def create_test_gateway_card(
```
Comment (bwiebe-solace) [inline tests/unit/common/test_gateway_registry.py:16]: These object creation methods looks similar (the same?) as the ones in test_a2a_utils.py. Might be beneficial to split these into their own test utils file for re-use.

File: src/solace_agent_mesh/common/base_registry.py (hunk 6 above)
Comment (mo-radwan1) [reply src/solace_agent_mesh/common/base_registry.py]: Fixed - updated the docstring to indicate that False is also returned if the card is invalid.

File: src/solace_agent_mesh/common/base_registry.py (hunk 7 above)
Comment (mo-radwan1) [reply src/solace_agent_mesh/common/base_registry.py]: Fixed - removed the unnecessary `removed` variable and now just return `True` directly.

File: src/solace_agent_mesh/common/base_registry.py (hunk 8 above)
Comment (mo-radwan1) [reply src/solace_agent_mesh/common/base_registry.py]: Good point - changed to INFO level since the caller (health check) already logs at WARNING when TTL expires. The remove() method is just executing what was requested. Also consolidated to a single log message.

File: src/solace_agent_mesh/common/base_registry.py (hunk 9 above)
Comment (mo-radwan1) [reply src/solace_agent_mesh/common/base_registry.py:220]: Good suggestion - added an INFO log that shows how many entities were cleared: `Cleared 5 gateway(s) from registry`. Only logs if count > 0.

File: src/solace_agent_mesh/services/platform/component.py (hunk 11 above)
Comment (mo-radwan1) [reply src/solace_agent_mesh/services/platform/component.py]: Fixed - moved the import to the top of the file with the other imports.

File: src/solace_agent_mesh/gateway/base/component.py (hunk 10 above)
Comment (mo-radwan1) [reply src/solace_agent_mesh/gateway/base/component.py:2312]: Good suggestion. For now with only 3-4 gateway types, the current if/chain is readable and works fine. If we add more gateway types in the future, I'll refactor this to use a mapping dict for better maintainability. Deferring for now to keep this PR focused.

File: tests/unit/common/test_gateway_registry.py (hunk 12 above)
Comment (mo-radwan1) [reply tests/unit/common/test_gateway_registry.py:16]: Agree these helpers are similar. For now I'm keeping them in each test file for isolation - test duplication is generally more acceptable than production code duplication. When we add more test files that need these helpers, I'll extract them to a shared conftest.py or test utils module.
